service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendAcknowledgementExecutor.java

```diff
@@ -0,0 +1,64 @@
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.mustachejava.Mustache;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.common.service.TimestampService;
+import uk.nhs.adaptors.gp2gp.common.task.TaskExecutor;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+import uk.nhs.adaptors.gp2gp.mhs.MhsClient;
+import uk.nhs.adaptors.gp2gp.mhs.MhsRequestBuilder;
+import uk.nhs.adaptors.gp2gp.mhs.model.OutboundMessageWithPayload;
+
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Service
+public class SendAcknowledgementExecutor implements TaskExecutor<SendAcknowledgementTaskDefinition> {
+    private static final Mustache ACKNOWLEDGEMENT_TEMPLATE =
+        TemplateUtils.loadTemplate("outbound_message_positive_acknowledgement.mustache");
+    private final MhsClient mhsClient;
+    private final MhsRequestBuilder mhsRequestBuilder;
+    private final EhrExtractStatusService ehrExtractStatusService;
+    private final ObjectMapper objectMapper;
+    private final TimestampService timestampService;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+
+    @Override
+    public Class<SendAcknowledgementTaskDefinition> getTaskType() {
+        return SendAcknowledgementTaskDefinition.class;
+    }
+
+    @Override
+    @SneakyThrows
+    public void execute(SendAcknowledgementTaskDefinition sendAcknowledgementTaskDefinition) {
+        LOGGER.info("SendAcknowledgement task was created, Sending Acknowledgement to Spine");
+        var messageId = randomIdGeneratorService.createNewId();
+
+        var sendAcknowledgementResponseObject = SendAcknowledgementResponse.builder()
+            .creationTime(timestampService.now().toString())
+            .uuid(messageId)
+            .fromAsid(sendAcknowledgementTaskDefinition.getFromAsid())
+            .toAsid(sendAcknowledgementTaskDefinition.getToAsid())
+            .typeCode(sendAcknowledgementTaskDefinition.getTypeCode())
+            .messageId(messageId)
+            .build();
+        var acknowledgmentRequestBody = TemplateUtils.fillTemplate(ACKNOWLEDGEMENT_TEMPLATE, sendAcknowledgementResponseObject);
+        var outboundMessage = OutboundMessageWithPayload.builder().payload(acknowledgmentRequestBody).build();
+        var stringRequestBody = objectMapper.writeValueAsString(outboundMessage);
+
+        var request = mhsRequestBuilder.buildSendAcknowledgement(stringRequestBody, sendAcknowledgementTaskDefinition.getFromOdsCode(),
+            sendAcknowledgementTaskDefinition.getConversationId());
+
+        mhsClient.sendMessageToMHS(request);
+
+        ehrExtractStatusService.updateEhrExtractStatusAcknowledgment(sendAcknowledgementTaskDefinition, messageId);
```

Comment:
This 'messageId' is the unique id of the message we just send over Spine via MHS. I expected the MHS Adaptor to generate and return a new Message ID in its response headers if one is not specific in the request. According to its API docs this is not the case: https://github.com/nhsconnect/integration-adaptor-mhs/blob/develop/mhs/outbound/mhsapi.json

We may need to modify MhsClient/Builder to always generate a new MessageId for the outbound request.